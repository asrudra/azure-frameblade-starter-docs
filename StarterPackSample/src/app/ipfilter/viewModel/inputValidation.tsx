export interface InputValidation {
    isValid: boolean;
    validationMessageKey?: string;
}