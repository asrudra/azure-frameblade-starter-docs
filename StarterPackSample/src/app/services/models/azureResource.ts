export interface AzureResource {
    resourceGroupName: string;
    resourceId: string;
    resourceName: string;
    subscriptionId: string;
}