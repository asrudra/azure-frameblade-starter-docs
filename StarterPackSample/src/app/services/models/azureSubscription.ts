export interface AzureSubscription {
    id: string;
}