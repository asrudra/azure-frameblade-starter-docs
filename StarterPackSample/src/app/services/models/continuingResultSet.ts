export interface ContinuingResultSet<T> {
    items: T[];
    continuationToken: string;
}