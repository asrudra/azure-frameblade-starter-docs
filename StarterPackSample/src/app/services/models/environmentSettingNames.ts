export enum EnvironmentSettingNames  {
    ARM_ENDPOINT = 'armEndpoint',
    AUTHORIZATION_TOKEN = 'authorizationToken',
    AZURE_RESOURCE = 'azureResource',
    LANGUAGE_CODE = 'languageCode',
    ROUTE_NAME = 'routeName'
}
