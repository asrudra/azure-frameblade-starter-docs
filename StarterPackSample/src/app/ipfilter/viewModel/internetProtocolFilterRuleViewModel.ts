import { IpFilterRule } from '@iotHubControlPlane/lib/models';
import { validateMask } from '../utilities/ipUtilities';
import { InputValidation } from './inputValidation';
import { ResourceKeys } from '../../../localization/resourceKeys';

export class InternetProtocolFilterRuleViewModel {
    public key: number;
    public filterRule: IpFilterRule;
    public nameValidation?: InputValidation;
    public addressRangeValidation?: InputValidation;

    constructor(filterRule: IpFilterRule, key: number) {
        this.filterRule = filterRule;
        this.key = key;
    }

    public static getDuplicates(filterRuleViewModels: InternetProtocolFilterRuleViewModel[]): Set<string> {
        const duplicateNames: Set<string> = new Set<string>();
        const allNames: Set<string> = new Set<string>();
    
        filterRuleViewModels.forEach((filterRuleViewModel) => {
            if (allNames.has(filterRuleViewModel.filterRule.filterName)) {
                duplicateNames.add(filterRuleViewModel.filterRule.filterName);
            } else {
                allNames.add(filterRuleViewModel.filterRule.filterName);
            }
        });
    
        return duplicateNames;
    }

    public validateName(duplicates: Set<string>) {

        if (!this.filterRule.filterName) {
            this.nameValidation = {
                isValid: false,
                validationMessageKey: ResourceKeys.ipFilter.filterName.validation.isBlank
            };
            return;
        }

        if (this.filterRule.filterName.length > 128) {
            this.nameValidation = {
                isValid: false,
                validationMessageKey: ResourceKeys.ipFilter.filterName.validation.exceedsAllowedLength
            };
            return;
        }

        const allowedCharacters = new RegExp('^[A-Za-z0-9-:.+%_#*?!(),=@;\']{1,128}$');
        if (!allowedCharacters.test(this.filterRule.filterName)) {
            this.nameValidation = {
                isValid: false,
                validationMessageKey: ResourceKeys.ipFilter.filterName.validation.hasInvalidCharacters
            };
            return;
        }

        if (duplicates.has(this.filterRule.filterName)) {
            this.nameValidation = {
                isValid: false,
                validationMessageKey: ResourceKeys.ipFilter.filterName.validation.isDuplicate
            };
            return;
        }

        this.nameValidation = {
            isValid: true
        };
    }

    public validateAddressRange() {

        if (!this.filterRule.ipMask) {
            this.addressRangeValidation = {
                isValid: false,
                validationMessageKey: ResourceKeys.ipFilter.ipMask.validation.isBlank
            };
            return;
        }

        if (this.filterRule.ipMask.length < 7) {
            this.addressRangeValidation = {
                isValid: false,
                validationMessageKey: ResourceKeys.ipFilter.ipMask.validation.isIncorrectLength
            };
            return;
        }

        if (this.filterRule.ipMask.length > 28) {
            this.addressRangeValidation = {
                isValid: false,
                validationMessageKey: ResourceKeys.ipFilter.ipMask.validation.isIncorrectLength
            };
            return;
        }

        if (!validateMask(this.filterRule.ipMask).isValid) {
            this.addressRangeValidation = {
                isValid: false,
                validationMessageKey: ResourceKeys.ipFilter.ipMask.validation.isIncorrectFormat
            };
            return;
        }

        this.addressRangeValidation = {
            isValid: true
        };
    }
}
