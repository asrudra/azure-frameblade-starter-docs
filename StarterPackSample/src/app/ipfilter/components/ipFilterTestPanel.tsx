import * as React from 'react';
import { PrimaryButton } from 'office-ui-fabric-react/lib/Button';
import { TextField } from 'office-ui-fabric-react/lib/TextField';
import { Panel, PanelType } from 'office-ui-fabric-react/lib/Panel';
import { Icon } from 'office-ui-fabric-react/lib/Icon';
import { TranslationFunction } from 'i18next';
import { IpFilterActionType } from '@iotHubControlPlane/lib/models';
import { checkIpAddressInRange, validateIpAddress } from '../utilities/ipUtilities';
import { ResourceKeys } from '../../../localization/resourceKeys';
import { IconNames } from '../../shared/utils/iconNames';
import { InternetProtocolFilterRuleViewModel } from '../viewModel/internetProtocolFilterRuleViewModel';

export interface IpFilterTestPanelProps {
    filterRuleViewModels: InternetProtocolFilterRuleViewModel[];
    onDismissClick: () => void;
    t: TranslationFunction;
}

interface IpFilterTestPanelState {
    affectedIpFilter: InternetProtocolFilterRuleViewModel | undefined;
    ipAddress: string;
    showResult: boolean;
    validationMessage: string;
}

export class IpFilterTestPanel extends React.Component<IpFilterTestPanelProps, IpFilterTestPanelState> {
    constructor(props: IpFilterTestPanelProps) {
        super(props);
        this.state = {
            affectedIpFilter: undefined,
            ipAddress: '',
            showResult: false,
            validationMessage: ''
        };
    }

    public render(): JSX.Element {
        return (
            <Panel
                className="iot-panel"
                isOpen={true}
                type={PanelType.medium}
                onDismiss={this.props.onDismissClick}
                headerText={this.props.t(ResourceKeys.ipFilter.testPanelMessages.headerText)}
            >
                {this.renderPanelContent()}
            </Panel>
        );
    }

    private renderPanelContent(): JSX.Element {
        return (
            <div>
                <p className="ipFilter-testPanel-description">
                    {this.props.t(ResourceKeys.ipFilter.testPanelMessages.description)}
                </p>
                <p className="ipFilter-testPanel-ipAddress-textField-label">
                    {this.props.t(ResourceKeys.ipFilter.testIpAddress.label)}
                </p>
                <TextField
                    className="ipFilter-testPanel-ipAddress-textField"
                    placeholder={this.props.t(ResourceKeys.ipFilter.testIpAddress.placeHolder)}
                    value={this.state.ipAddress}
                    onChanged={this.onChangeHandler}
                    errorMessage={this.state.validationMessage}
                />
                {this.renderButtonsContent()}
                <div className="ipFilter-testPanel-results">
                    {this.state.showResult && this.renderTestResult()}
                </div>
            </div>
        );
    }

    private renderTestResult(): JSX.Element {
        const { t } = this.props;
        const { affectedIpFilter } = this.state;
        if (!affectedIpFilter) {
            return (<p>{this.props.t(ResourceKeys.ipFilter.testPanelMessages.noMatch)}</p>);
        }

        const action = affectedIpFilter.filterRule.action === IpFilterActionType.Accept ? t(ResourceKeys.ipFilter.testPanelMessages.acceptedAction) : t(ResourceKeys.ipFilter.testPanelMessages.rejectedAction);
        return (
            <>
                {this.renderMatchFoundMessage(action)}
                {this.renderRowResult(affectedIpFilter)}
            </>
        );
    }

    private renderMatchFoundMessage(action: string): JSX.Element {
        const { t } = this.props;
        return (
            <p className="ipFilter-testPanel-results-match-found-message">
                <span>{t(ResourceKeys.ipFilter.testPanelMessages.matches.prefix)}</span>
                <span className="ipFilter-testPanel-results-action-name">{` ${action} `}</span>
                <span>{t(ResourceKeys.ipFilter.testPanelMessages.matches.suffix)}</span>
            </p>
        );
    }

    private renderRowResult(filterRuleViewModel: InternetProtocolFilterRuleViewModel): JSX.Element {
        const actionIcon = filterRuleViewModel.filterRule.action === IpFilterActionType.Accept ?
            <Icon iconName={IconNames.Allow} className="ipFilter-testPanel-results-allow-icon"/> :
            <Icon iconName={IconNames.Block} className="ipFilter-testPanel-results-block-icon"/>;

        return (
            <p>
                <span className="ipFilter-testPanel-results-icon">{actionIcon}</span>
                <span className="ipFilter-testPanel-results-filter-name">{filterRuleViewModel.filterRule.filterName}</span>
                <span className="ipFilter-testPanel-results-ipMask">{`(${filterRuleViewModel.filterRule.ipMask})`}</span>
            </p>
        );
    }

    private renderButtonsContent(): JSX.Element {
        const { t } = this.props;
        const { validationMessage, ipAddress } = this.state;
        return (
            <div className="ipFilter-testPanel-buttons">
                <PrimaryButton
                    onClick={this.verifyClickHandler}
                    className="ipFilter-testPanel-verify-button"
                    disabled={!!validationMessage || !ipAddress}
                >
                    {t(ResourceKeys.ipFilter.testVerifyButton.name)}
                </PrimaryButton>
            </div>
        );
    }

    private onChangeHandler = (value: string): void => {
        const validationMessage = (!value || validateIpAddress(value)) ? '' : this.props.t(ResourceKeys.ipFilter.testIpAddress.errorMessage);
        this.setState({
            ipAddress: value,
            validationMessage
        });
    }

    private verifyClickHandler = (): void => {
        const affectedIpFilters = this.props.filterRuleViewModels.filter((model) => checkIpAddressInRange(this.state.ipAddress, model.filterRule.ipMask));
        const affectedIpFilter = affectedIpFilters && affectedIpFilters.pop();
        this.setState({
            affectedIpFilter,
            showResult: true
        });
    }
}
