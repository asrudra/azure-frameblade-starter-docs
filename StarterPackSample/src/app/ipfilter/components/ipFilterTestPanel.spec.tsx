import 'jest';
import context from 'jest-plugin-context';
import * as React from 'react';
import { shallow } from 'enzyme';
import { IpFilterRule, IpFilterActionType } from '@iotHubControlPlane/lib/models';
import { IpFilterTestPanel } from './ipFilterTestPanel';
import { InternetProtocolFilterRuleViewModel } from '../viewModel/internetProtocolFilterRuleViewModel';

let mockTranslationFunction;
let translationMap;

beforeAll(() => {
    translationMap = {
        'ipFilter.testIpAddress.errorMessage': 'validation error',
        'ipFilter.testPanelMessages.acceptedAction': 'allowed',
        'ipFilter.testPanelMessages.matches.prefix': 'This IP address is',
        'ipFilter.testPanelMessages.matches.suffix': 'by the following filter:',
        'ipFilter.testPanelMessages.noMatch': 'No Result Found',
        'ipFilter.testPanelMessages.rejectedAction': 'blocked'
    };

    mockTranslationFunction = (input: string) => {
        return translationMap[input];
    };
});

afterAll(() => {
    translationMap = undefined;
    mockTranslationFunction = undefined;
});

describe('ipFilter/components/ipFilterTestPanel', () => {
    context('snapshot', () => {
        it('matches snapshot', () => {
            const wrapper = shallow(<IpFilterTestPanel onDismissClick={jest.fn()} t={jest.fn()} filterRuleViewModels={[]}/>);

            expect(wrapper).toMatchSnapshot();
        });
    });

    context('state/initial', () => {
        it('should initialize state properly', () => {
            const wrapper = shallow(<IpFilterTestPanel onDismissClick={jest.fn()} t={jest.fn()} filterRuleViewModels={[]}/>);

            expect(wrapper.state()).toEqual({
                affectedIpFilter: undefined,
                ipAddress: '',
                showResult: false,
                validationMessage: ''
            });
        });
    });

    context('state/ipAddress', () => {
        it('should change based on user input', () => {
            const wrapper = shallow(<IpFilterTestPanel onDismissClick={jest.fn()} t={jest.fn()} filterRuleViewModels={[]}/>);
            let state = wrapper.state() as any; //tslint:disable-line
            expect(state.ipAddress).toEqual('');

            const textInput = wrapper.find('.ipFilter-testPanel-ipAddress-textField');
            textInput.simulate('changed', '123');

            state = wrapper.state();
            expect(state.ipAddress).toEqual('123');

            textInput.simulate('changed', '123.123.123.1');

            state = wrapper.state();
            expect(state.ipAddress).toEqual('123.123.123.1');
        });
    });

    context('state/validationMessage', () => {
        it('should add/remove validation message based on whether ip address input is valid or not', () => {
            const wrapper = shallow(<IpFilterTestPanel onDismissClick={jest.fn()} t={mockTranslationFunction} filterRuleViewModels={[]}/>);
            let state = wrapper.state() as any; //tslint:disable-line
            expect(state.validationMessage).toEqual('');

            const textInput = wrapper.find('.ipFilter-testPanel-ipAddress-textField');
            textInput.simulate('changed', '123');

            state = wrapper.state();
            expect(state.validationMessage).toEqual('validation error');

            textInput.simulate('changed', '123.123.123.1');

            state = wrapper.state();
            expect(state.validationMessage).toEqual('');
        });
    });

    context('render/verifyButton', () => {
        it('disables/enables verify button based on whether ip address input is valid or not', () => {
            const wrapper = shallow(<IpFilterTestPanel onDismissClick={jest.fn()} t={mockTranslationFunction} filterRuleViewModels={[]}/>);

            const textInput = wrapper.find('.ipFilter-testPanel-ipAddress-textField');
            let verifyButton = wrapper.find('.ipFilter-testPanel-verify-button').first();

            expect(verifyButton.props().disabled).toBeTruthy();

            textInput.simulate('changed', '123.123.123.1');
            wrapper.update();
            verifyButton = wrapper.find('.ipFilter-testPanel-verify-button').first();
            expect(verifyButton.props().disabled).toBeFalsy();

            textInput.simulate('changed', '123');
            wrapper.update();
            verifyButton = wrapper.find('.ipFilter-testPanel-verify-button').first();
            expect(verifyButton.props().disabled).toBeTruthy();
        });
    });

    context('render/testResults', () => {
        it('shows no result message for valid ip address input on verify button click when none of the filter filterRules are affected', () => {
            const wrapper = shallow(<IpFilterTestPanel onDismissClick={jest.fn()} t={mockTranslationFunction} filterRuleViewModels={[]}/>);

            let results = wrapper.find('.ipFilter-testPanel-results');

            expect(results.text()).toEqual('');

            wrapper.find('.ipFilter-testPanel-ipAddress-textField').simulate('changed', '123.123.123.1');
            wrapper.update();
            wrapper.find('.ipFilter-testPanel-verify-button').first().simulate('click');
            wrapper.update();
            results = wrapper.find('.ipFilter-testPanel-results');
            expect(results.text()).toEqual('No Result Found');
            expect(wrapper).toMatchSnapshot();
        });

        it('shows results for valid ip address input on verify button click when given ip address is blocked by any of the filter rules', () => {
            const mockFilterRuleViewModels: InternetProtocolFilterRuleViewModel[] = [
                new InternetProtocolFilterRuleViewModel(
                    {
                        action: IpFilterActionType.Accept,
                        filterName: 'filter_one',
                        ipMask: '127.127.123.1/28'
                    },
                    1
                ),
                new InternetProtocolFilterRuleViewModel(
                    {
                        action: IpFilterActionType.Reject,
                        filterName: 'filter_two',
                        ipMask: '127.127.123.2/20'
                    },
                    2
                )
            ];

            const wrapper = shallow(<IpFilterTestPanel onDismissClick={jest.fn()} t={mockTranslationFunction} filterRuleViewModels={mockFilterRuleViewModels}/>);

            const results = wrapper.find('.ipFilter-testPanel-results');

            expect(results.text()).toEqual('');

            wrapper.find('.ipFilter-testPanel-ipAddress-textField').simulate('changed', '127.127.123.6');
            wrapper.update();
            wrapper.find('.ipFilter-testPanel-verify-button').first().simulate('click');
            wrapper.update();

            expect(wrapper.find('.ipFilter-testPanel-results-match-found-message').text()).toEqual('This IP address is blocked by the following filter:');
            expect(wrapper.find('.ipFilter-testPanel-results-filter-name').text()).toEqual('filter_two');
            expect(wrapper.find('.ipFilter-testPanel-results-ipMask').text()).toEqual('(127.127.123.2/20)');
            expect(wrapper).toMatchSnapshot();
        });

        it('shows results for valid ip address input on verify button click when given ip address is allowed by any of the filter rules', () => {
            const mockFilterRuleViewModels: InternetProtocolFilterRuleViewModel[] = [
                new InternetProtocolFilterRuleViewModel(
                    {
                        action: IpFilterActionType.Reject,
                        filterName: 'filter_two',
                        ipMask: '127.127.123.2/20'
                    },
                    2
                ),
                new InternetProtocolFilterRuleViewModel(
                    {
                        action: IpFilterActionType.Accept,
                        filterName: 'filter_one',
                        ipMask: '127.127.123.1/28'
                    },
                    1
                )
            ];

            const wrapper = shallow(<IpFilterTestPanel onDismissClick={jest.fn()} t={mockTranslationFunction} filterRuleViewModels={mockFilterRuleViewModels}/>);

            const results = wrapper.find('.ipFilter-testPanel-results');

            expect(results.text()).toEqual('');

            wrapper.find('.ipFilter-testPanel-ipAddress-textField').simulate('changed', '127.127.123.6');
            wrapper.update();
            wrapper.find('.ipFilter-testPanel-verify-button').first().simulate('click');
            wrapper.update();

            expect(wrapper.find('.ipFilter-testPanel-results-match-found-message').text()).toEqual('This IP address is allowed by the following filter:');
            expect(wrapper.find('.ipFilter-testPanel-results-filter-name').text()).toEqual('filter_one');
            expect(wrapper.find('.ipFilter-testPanel-results-ipMask').text()).toEqual('(127.127.123.1/28)');
            expect(wrapper).toMatchSnapshot();
        });
    });
});
