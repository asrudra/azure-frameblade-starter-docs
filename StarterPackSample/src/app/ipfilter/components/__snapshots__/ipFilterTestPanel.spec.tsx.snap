// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ipFilter/components/ipFilterTestPanel render/testResults shows no result message for valid ip address input on verify button click when none of the filter filterRules are affected 1`] = `
<StyledPanelBase
  className="iot-panel"
  isOpen={true}
  onDismiss={[MockFunction]}
  type={3}
>
  <div>
    <p
      className="ipFilter-testPanel-description"
    />
    <p
      className="ipFilter-testPanel-ipAddress-textField-label"
    />
    <StyledTextFieldBase
      className="ipFilter-testPanel-ipAddress-textField"
      errorMessage=""
      onChanged={[Function]}
      value="123.123.123.1"
    />
    <div
      className="ipFilter-testPanel-buttons"
    >
      <CustomizedPrimaryButton
        className="ipFilter-testPanel-verify-button"
        disabled={false}
        onClick={[Function]}
      />
    </div>
    <div
      className="ipFilter-testPanel-results"
    >
      <p>
        No Result Found
      </p>
    </div>
  </div>
</StyledPanelBase>
`;

exports[`ipFilter/components/ipFilterTestPanel render/testResults shows results for valid ip address input on verify button click when given ip address is allowed by any of the filter rules 1`] = `
<StyledPanelBase
  className="iot-panel"
  isOpen={true}
  onDismiss={[MockFunction]}
  type={3}
>
  <div>
    <p
      className="ipFilter-testPanel-description"
    />
    <p
      className="ipFilter-testPanel-ipAddress-textField-label"
    />
    <StyledTextFieldBase
      className="ipFilter-testPanel-ipAddress-textField"
      errorMessage=""
      onChanged={[Function]}
      value="127.127.123.6"
    />
    <div
      className="ipFilter-testPanel-buttons"
    >
      <CustomizedPrimaryButton
        className="ipFilter-testPanel-verify-button"
        disabled={false}
        onClick={[Function]}
      />
    </div>
    <div
      className="ipFilter-testPanel-results"
    >
      <p
        className="ipFilter-testPanel-results-match-found-message"
      >
        <span>
          This IP address is
        </span>
        <span
          className="ipFilter-testPanel-results-action-name"
        >
           allowed 
        </span>
        <span>
          by the following filter:
        </span>
      </p>
      <p>
        <span
          className="ipFilter-testPanel-results-icon"
        >
          <StyledIconBase
            className="ipFilter-testPanel-results-allow-icon"
            iconName="SkypeCircleCheck"
          />
        </span>
        <span
          className="ipFilter-testPanel-results-filter-name"
        >
          filter_one
        </span>
        <span
          className="ipFilter-testPanel-results-ipMask"
        >
          (127.127.123.1/28)
        </span>
      </p>
    </div>
  </div>
</StyledPanelBase>
`;

exports[`ipFilter/components/ipFilterTestPanel render/testResults shows results for valid ip address input on verify button click when given ip address is blocked by any of the filter rules 1`] = `
<StyledPanelBase
  className="iot-panel"
  isOpen={true}
  onDismiss={[MockFunction]}
  type={3}
>
  <div>
    <p
      className="ipFilter-testPanel-description"
    />
    <p
      className="ipFilter-testPanel-ipAddress-textField-label"
    />
    <StyledTextFieldBase
      className="ipFilter-testPanel-ipAddress-textField"
      errorMessage=""
      onChanged={[Function]}
      value="127.127.123.6"
    />
    <div
      className="ipFilter-testPanel-buttons"
    >
      <CustomizedPrimaryButton
        className="ipFilter-testPanel-verify-button"
        disabled={false}
        onClick={[Function]}
      />
    </div>
    <div
      className="ipFilter-testPanel-results"
    >
      <p
        className="ipFilter-testPanel-results-match-found-message"
      >
        <span>
          This IP address is
        </span>
        <span
          className="ipFilter-testPanel-results-action-name"
        >
           blocked 
        </span>
        <span>
          by the following filter:
        </span>
      </p>
      <p>
        <span
          className="ipFilter-testPanel-results-icon"
        >
          <StyledIconBase
            className="ipFilter-testPanel-results-block-icon"
            iconName="BlockedSolid"
          />
        </span>
        <span
          className="ipFilter-testPanel-results-filter-name"
        >
          filter_two
        </span>
        <span
          className="ipFilter-testPanel-results-ipMask"
        >
          (127.127.123.2/20)
        </span>
      </p>
    </div>
  </div>
</StyledPanelBase>
`;

exports[`ipFilter/components/ipFilterTestPanel snapshot matches snapshot 1`] = `
<StyledPanelBase
  className="iot-panel"
  isOpen={true}
  onDismiss={[MockFunction]}
  type={3}
>
  <div>
    <p
      className="ipFilter-testPanel-description"
    />
    <p
      className="ipFilter-testPanel-ipAddress-textField-label"
    />
    <StyledTextFieldBase
      className="ipFilter-testPanel-ipAddress-textField"
      errorMessage=""
      onChanged={[Function]}
      value=""
    />
    <div
      className="ipFilter-testPanel-buttons"
    >
      <CustomizedPrimaryButton
        className="ipFilter-testPanel-verify-button"
        disabled={true}
        onClick={[Function]}
      />
    </div>
    <div
      className="ipFilter-testPanel-results"
    />
  </div>
</StyledPanelBase>
`;
